src/wazuh_modules/wm_sca.c

```diff
@@ -298,114 +298,24 @@ static void wm_sca_send_policies_scanned(wm_sca_t * data) {
 
 static int wm_sca_start(wm_sca_t * data) {
 
-    int status = 0;
-    time_t time_start = 0;
-    time_t time_sleep = 0;
-
-    if (!data->scan_on_start) {
-        time_start = time(NULL);
-
-        if (data->scan_day) {
-            do {
-                status = check_day_to_scan(data->scan_day, data->scan_time);
-                if (status == 0) {
-                    time_sleep = get_time_to_hour(data->scan_time);
-                } else {
-                    wm_delay(1000); // Sleep one second to avoid an infinite loop
-                    time_sleep = get_time_to_hour("00:00");
-                }
-
-                mdebug2("Sleeping for %d seconds.", (int)time_sleep);
-                wm_delay(1000 * time_sleep);
-
-            } while (status < 0);
-
-        } else if (data->scan_wday >= 0) {
-
-            time_sleep = get_time_to_day(data->scan_wday, data->scan_time);
-            minfo("Waiting for turn to evaluate.");
-            mdebug2("Sleeping for %d seconds.", (int)time_sleep);
-            wm_delay(1000 * time_sleep);
-
-        } else if (data->scan_time) {
-
-            time_sleep = get_time_to_hour(data->scan_time);
-            minfo("Waiting for turn to evaluate.");
-            mdebug2("Sleeping for %d seconds.", (int)time_sleep);
-            wm_delay(1000 * time_sleep);
-
-        } else if (data->next_time == 0 || data->next_time > time_start) {
-
-            // On first run, take into account the interval of time specified
-            time_sleep = data->next_time == 0 ?
-                         (time_t)data->interval :
-                         data->next_time - time_start;
-
-            minfo("Waiting for turn to evaluate.");
-            mdebug2("Sleeping for %ld seconds.", (long)time_sleep);
+    do {
+        const time_t time_sleep = sched_scan_get_next_time(&(data->scan_config), WM_GCP_LOGTAG, data->scan_on_start);
+        
+        if (time_sleep) {
+            mtdebug1(WM_SCA_LOGTAG, "Sleeping for %li seconds", time_sleep);
             wm_delay(1000 * time_sleep);
-
         }
-    }
-
-    while(1) {
-        // Get time and execute
-        time_start = time(NULL);
-
-        minfo("Starting Security Configuration Assessment scan.");
+        mtdebug1(WM_SCA_LOGTAG,"Starting Security Configuration Assessment scan.");
```

Comment:
Why was this log message changed from info to debug1 level?

Reply:
Just to match with other modules that are using the same log level in the run loop. There is already a INFO message indicating that the module is starting before entering the loop.

Reply:
Hi guys, I think it should keep going at info level. 
That way, the user knows when the scan is launched. Here some examples of the same log in other modules: https://github.com/wazuh/wazuh/blob/971b1acbb881c86b20c0d364ce12f004ec43419d/src/syscheckd/run_check.c#L80
https://github.com/wazuh/wazuh/blob/971b1acbb881c86b20c0d364ce12f004ec43419d/src/wazuh_modules/syscollector/syscollector_common.c#L76
Regards.

Reply:
Ok great, I will repeat the same for the rest of the wazuh_modules then since most of them are on `debug` level